Add controllers for topics and user-topics

aeb0 opened the pull request:
Created handler functions to add, remove, and get topics from topics table.
Created handler functions to add, remove and get user topics from table.
Added routes for each controller, and included those routes in top-level router.

gnawwaycon committed:
change twitter model so no messages are stored and begun refactoring the wtitter worker to fit this model

gnawwaycon committed:
make twitterdata grab work for set array

gnawwaycon committed:
add twitter stream packages to json and fix sequelize exports

gnawwaycon committed:
add twitter stream package

gnawwaycon committed:
filter only english results

Commit:
Add controllers for topics and user-topics
Created handler functions to add, remove, and get topics from topics table.
Created handler functions to add, remove and get user topics from table.
Added routes for each controller, and included those routes in top-level router.

Commit:
Add controllers for topics and user-topics
Created handler functions to add, remove, and get topics from topics table.
Created handler functions to add, remove and get user topics from table.
Added routes for each controller, and included those routes in top-level router.

aeb0 commented on server/config/passport.js:
```diff
@@ -19,6 +19,8 @@ passport.use(new JwtStrategy(
       .then(user => {
         if (user) {
+          //see if this still works if comment out req.user = user (i.e. it is done automatically)
+          req.user = user;
```
delete this sonbitc

aeb0 commented on server/topics/topic.routes.js:
```diff
@@ -0,0 +1,11 @@
+
+
+TopicRouter.post('/addTopic', TopicCtrl.addTopic, UserTopicsCtrl.addUserTopic);
+TopicRouter.get('/getAllTopics', TopicCtrl.getAllTopics);
```
no need

aeb0 commented on server/topics/topic.controller.js:
```diff
@@ -0,0 +1,40 @@
+  .catch(err => console.error(`Error writing topic to db: ${err}`));
+}
+
+function getAllTopics(req, res) {
```
no need

aeb0 commented on server/user-topics/user-topics.controller.js:
```diff
@@ -0,0 +1,56 @@
+      UserTopic.create({
+        topicId: req.topic.topicId,
+        userId: req.topic.userId,
+      })
```
in the then res.send something since this is ifnal middleware (res.sendStatus(201)

aeb0 replied:
also do it in the catch - sendstatus 500

aeb0 commented on server/topics/topic.controller.js:
```diff
@@ -0,0 +1,40 @@
+}
+
+function removeTopic(req, res) {
+  Topic.destroy({
```
need to send status here

aebrow4 commented on server/user-topics/user-topics.controller.js:
```diff
@@ -0,0 +1,56 @@
+  UserTopic.find({
+    where: {
+      // pull userId off the token?
+      userId: req.user.id
```
respond

aeb0 commented on server/user-topics/user-topics.controller.js:
```diff
@@ -0,0 +1,56 @@
+  });
+}
+
+function getAllUserTopics(req, res) {
```
respond

aeb0 commented on server/user-topics/user-topics.controller.js:
```diff
@@ -0,0 +1,56 @@
+      topicId: req.body.topicId,
+    }
+  });
+  UserTopic.find({
```
then...
also find always returns an array so check length if using truthiness

aeb0 commented on server/user-topics/user-topics.controller.js:
```diff
@@ -0,0 +1,56 @@
+    .then(userTopics => {
+      if(!userTopics) {
+        next(null);
+      }
```
else respond since we are not going to another middlware (and in the catch - 500)

aebrow4 commented on server/user-topics/user-topics.routes.js:
```diff
@@ -0,0 +1,9 @@
+const UserTopicCtrl = require('./user-topics.controller');
+const TopicCtrl = require('../topics/topic.model');
+
+UserTopicsRouter.post('/addTopic', UserTopicCtrl.addUserTopic);
```
add the TopicCtrl add middleware

aeb0 replied:
dont need this route since UserTopics add function will be called in the middleware chain of topics router

aeb0 commented on server/user-topics/user-topics.routes.js:
```diff
@@ -0,0 +1,9 @@
+const TopicCtrl = require('../topics/topic.model');
+
+UserTopicsRouter.post('/addTopic', UserTopicCtrl.addUserTopic);
+UserTopicsRouter.get('/getAllTopics', UserTopicCtrl.getAllUserTopics);
```
noun is not necessary

aeb0 commented on server/topics/topic.routes.js:
```diff
@@ -0,0 +1,11 @@
+const UserTopicsCtrl = require('../user-topics.controller');
+
+
+TopicRouter.post('/addTopic', TopicCtrl.addTopic, UserTopicsCtrl.addUserTopic);
```
noun not necessary

gnawwaycon committed:
remove extraneous comments

ssuksawat committed:
Merge pull request #18 from gnawwaycon/feat/twitterGrab

Feat/twitter grab

Commit:
Made changes based on PR review
Modified router paths to not include nouns.
Inserted calls to next in certain request handlers where another middleware function needed to follow
Included res.sendStatus in some places where another middleware function was not being called

Commit:
Resolved merge conflicts
Merge branch 'feature-Users' of https://github.com/aebrow4/desultoryhalibut into feature-Users

Commit:
Resolved merge conflict
